clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1688,15 +1691,27 @@ void handleFailure(Throwable throwable) {
     }
 
     @Override
-    public DeleteTopicsResult deleteTopics(final Collection<String> topicNames,
+    public DeleteTopicsResult deleteTopics(final TopicCollection topics,
                                            final DeleteTopicsOptions options) {
+        DeleteTopicsResult result;
+        if (topics instanceof TopicIdCollection)
+            result = DeleteTopicsResult.ofTopicIds(new HashMap<>(handleDeleteTopicsUsingIds(((TopicIdCollection) topics).topicIds(), options)));
```

Comment:
Why do we copy the result of `handleDeleteTopicsUsingIds`? Seems like that method is already returning a fresh map.

Reply:
This was an artifact of the old version of the code. We would create a copy of the map 
```
return new DeleteTopicsResult(new HashMap<>(topicFutures));
```
I can remove though.

Reply:
Hmmm. When I try to remove I get
`'ofTopicIds(java.util.Map<org.apache.kafka.common.Uuid,org.apache.kafka.common.KafkaFuture<java.lang.Void>>)' in 'org.apache.kafka.clients.admin.DeleteTopicsResult' cannot be applied to '(java.util.Map<org.apache.kafka.common.Uuid,org.apache.kafka.common.internals.KafkaFutureImpl<java.lang.Void>>)'`

Reply:
All other admin client calls also follow this pattern with using KafkaFutureImpl and copying the result to get KafkaFutures

Reply:
I can move this cast to the `handle...` call and return a KafkaFuture if it makes things cleaner.